htlcswitch/failure.go

```diff
@@ -74,10 +74,6 @@ type ForwardingError struct {
 	// zero is the self node.
 	FailureSourceIdx int
 
-	// ExtraMsg is an additional error message that callers can provide in
-	// order to provide context specific error details.
-	ExtraMsg string
```

Comment:
was this even used anywhere?

Reply:
It was used in [handleLocalDispatch](https://github.com/lightningnetwork/lnd/blob/3c6be62b18b9ea84369ac92d48ebcd2325807d3c/htlcswitch/switch.go#L775) to set a link not available error and in [parseFailedPayment](https://github.com/lightningnetwork/lnd/blob/3c6be62b18b9ea84369ac92d48ebcd2325807d3c/htlcswitch/switch.go#L935) if we could not decode the onion, or the payment went on chain before the first hop.

These are replaced with `FailureDetail`, so the error presented to the user no longer has the exact link that failed/ error that occurred, but the full original error is logged and I doubt anybody has been parsing onion decode payment errors for details.

Reply:
👍 

Could be removed in the same commit that replaces it with `FailureDetail`, but that is no blocker :)